### PR TITLE
Create post button patch

### LINKED ISSUE #40: Create Post Button Outdated
**Description of the issue**
The current create post button does not properly result in a new post when correct information is inputted by the user.

**Files that need to be updated**
The files that need to be changed or refactored are posts/buttons/create-post-button.tsx and posts/pages/create-post.tsx. These components are mainly involved in creating posts, so they need to be changed. 

**Link**
Include a link to the correlated PR number, or any resources for greater context.

**Additional context**
The update #30 to Posts in the backend caused the existing create post components to become outdated.

### PR DESCRIPTION
**Pull request by thecatfour**
### Category
- [ ] **Feature**
- [x] **Bug Fix**
- [ ] **Testing**
- [ ] **Documentation**
- [ ] **Other**

### Domain
- [ ] **Frontend**
- [ ] **Backend**
- [x] **Both**

### Description
This allows posts to be created in the frontend by creating two new components to address the updated Posts backend.

### Related Issue(s)
This PR fixes #40 by creating a new form component and a new helper component.

### New Additions
**create-post-form.tsx**
This component provides and submits a form with all the necessary information to create a post.

**hub-input.tsx**
This component provides a select drop menu to allow the user to select a hub that they have subscribed to. This means they can select from the drop menu rather than typing in the hub.

### Modifications

- **pages/create-post.tsx:** Now simply calls the CreatePostForm component.
- **definitions.ts:** Added the Hub type.

### New Dependencies
There are no new dependencies.

### Visuals
This is a simple gif showing the CreatePostForm component:

ezgif-3-183b2bca53

### Architecture of Your Changes
**CreatePostForm**
This component both creates the form necessary to create a post as well as makes the API call to the backend. It uses several helper functions such as create-post-input.tsx and hub-input.tsx to gather and validate input.

**hub-input.tsx**
This component makes a select drop menu of hubs that the user has subscribed to. It probably will not have any use in any other context.

**definitions.ts**
The Hub type was added to help simplify the parameters in some components.

### Usage and Integration
This PR adds the CreatePostForm component, which can be called to create a form to create a post. In the future, this could be used to implement post creation without an explicit create post page.

### Testing
The changes were simply tested using the dev website from Django and pnpm. The following gif shows some of the basic input validation that the form component currently does:

ezgif-3-c071d506de

Basically, a title and message are required. If a user is not subscribed to any hubs, the drop down menu will instead display a message rather than hub options.

### Mentions
These changes are not currently particularly useful to anyone as hubs have not been implemented in the frontend yet. In the future, @imawful and @misherwa may need these to create posts in specific hubs or to develop the frontend for comments.

### CODE REVIEW

**Comment by thecatfour** [comment]
Note: This PR is based on my like/dislike branch, so merging this branch will also merge the likes/dislikes. #38 should be merged first to make it clear which PR is changing what.

**Review by nnguyen144 — APPROVED** [review]
I've tested creating a post with your code and was successfully able to, so for now your code satisfies the issue of creating posts. That being said, there are things that you could add when creating a post, such as image upload, tags, and maybe polls. But for now, this code is approved (after the merge conflict is fixed).

**Review by courtmcgee2197 — APPROVED** [review]
I tested the functionality of creating a post and it works. I do think for the future we should be able to add images to a post. Also visually its pretty dark compared to our prototype home page we may have to agree on an overall color scheme of our website

**Pull request merged**
